src/framework/uima/internal_engine_base.hpp

```diff
@@ -64,6 +64,10 @@ namespace uima {
 /*       Types / Classes                                                   */
 /* ----------------------------------------------------------------------- */
 namespace uima {
+
```

Comment:
So there's no other exception already defined by the framework that we could use? The framework already declares other 80 exceptions:

```
uima-uimacpp/src$ find . -name \*.hpp -exec grep -H  UIMA_EXC_CLASSDECLARE \{} \;
./cas/uima/featurestructure.hpp:  UIMA_EXC_CLASSDECLARE(InvalidFSObjectException, CASException);
./cas/uima/featurestructure.hpp:  UIMA_EXC_CLASSDECLARE(FeatureNotAppropriateException, CASException);
./cas/uima/featurestructure.hpp:  UIMA_EXC_CLASSDECLARE(IncompatibleValueTypeException, CASException);
./cas/uima/featurestructure.hpp:  UIMA_EXC_CLASSDECLARE(FSIsNotStringException, CASException);
./cas/uima/featurestructure.hpp:  UIMA_EXC_CLASSDECLARE(WrongStringValueException, CASException);
./cas/uima/arrayfs.hpp:  UIMA_EXC_CLASSDECLARE(FSIsNotArrayException, CASException);
./cas/uima/arrayfs.hpp:  UIMA_EXC_CLASSDECLARE(FSArrayOutOfBoundsException, CASException);
./cas/uima/fsindexrepository.hpp:  UIMA_EXC_CLASSDECLARE(InvalidIndexIDException, CASException);
./cas/uima/typesystem.hpp:  UIMA_EXC_CLASSDECLARE(TypeSystemAlreadyCommittedException, CASException);
./cas/uima/typesystem.hpp:  UIMA_EXC_CLASSDECLARE(InvalidFSFeatureObjectException, CASException);
./cas/uima/typesystem.hpp:  UIMA_EXC_CLASSDECLARE(InvalidFSTypeObjectException, CASException);
./cas/uima/casexception.hpp:  UIMA_EXC_CLASSDECLARE(CASException, Exception);
./cas/uima/casexception.hpp:  UIMA_EXC_CLASSDECLARE(NotYetImplementedException, CASException);
./cas/uima/casexception.hpp:  UIMA_EXC_CLASSDECLARE(SofaDataStreamException, CASException);
./cas/uima/cas.hpp:  UIMA_EXC_CLASSDECLARE(CouldNotCreateFSOfFinalTypeException, CASException);
./cas/uima/cas.hpp:  UIMA_EXC_CLASSDECLARE(DuplicateSofaNameException, CASException);
./cas/uima/cas.hpp:  UIMA_EXC_CLASSDECLARE(InvalidBaseCasMethod, CASException);
./cas/uima/xmltypesystemreader.hpp:  UIMA_EXC_CLASSDECLARE(XMLTypeSystemReaderException, uima::Exception);
./cas/uima/lowlevel_typesystem.hpp:    UIMA_EXC_CLASSDECLARE(FeatureIntroductionFailedException, CASException);
./cas/uima/lowlevel_typesystem.hpp:    UIMA_EXC_CLASSDECLARE(TypeCreationFailedException, CASException);
./cas/uima/fsindex.hpp:  UIMA_EXC_CLASSDECLARE(InvalidIndexObjectException, CASException);
./cas/uima/fsindex.hpp:  UIMA_EXC_CLASSDECLARE(WrongFSTypeForIndexException, CASException);
./cas/uima/listfs.hpp:  UIMA_EXC_CLASSDECLARE(FSIsNotListException, CASException);
./cas/uima/listfs.hpp:  UIMA_EXC_CLASSDECLARE(ListIsEmptyException, CASException);
./cas/uima/listfs.hpp:  UIMA_EXC_CLASSDECLARE(ListIsCircularException, CASException);
./cas/uima/internal_casdeserializer.hpp:    UIMA_EXC_CLASSDECLARE(DeserializationException, Exception);
./framework/uima/exceptions.hpp:   UIMA_EXCEPTION_DESCRIPTION, UIMA_EXC_CLASSDECLARE, UIMA_EXC_CLASSIMPLEMENT,
./framework/uima/exceptions.hpp:     <TT>UIMA_EXC_CLASSDECLARE</TT> in an .h/.hpp file there must be one use of
./framework/uima/exceptions.hpp:#define UIMA_EXC_CLASSDECLARE(child,parent)                                    \
./framework/uima/exceptions.hpp:     <TT>UIMA_EXC_CLASSDECLARE</TT> (see above) in the corresponding .h/.hpp file
./framework/uima/exceptions.hpp:  UIMA_EXC_CLASSDECLARE(Uima_logic_error,      Exception);
./framework/uima/exceptions.hpp:  UIMA_EXC_CLASSDECLARE(Uima_runtime_error,    Exception);
./framework/uima/exceptions.hpp:  UIMA_EXC_CLASSDECLARE(Uima_domain_error,     Uima_logic_error);
./framework/uima/exceptions.hpp:  UIMA_EXC_CLASSDECLARE(Uima_invalid_argument, Uima_logic_error);
./framework/uima/exceptions.hpp:  UIMA_EXC_CLASSDECLARE(Uima_length_error,     Uima_logic_error);
./framework/uima/exceptions.hpp:  UIMA_EXC_CLASSDECLARE(Uima_out_of_range,     Uima_logic_error);
./framework/uima/exceptions.hpp:  UIMA_EXC_CLASSDECLARE(Uima_range_error,      Uima_runtime_error);
./framework/uima/exceptions.hpp:  UIMA_EXC_CLASSDECLARE(Uima_overflow_error,   Uima_runtime_error);
./framework/uima/exceptions.hpp:  UIMA_EXC_CLASSDECLARE(Uima_underflow_error,  Uima_runtime_error);
./framework/uima/exceptions.hpp:  UIMA_EXC_CLASSDECLARE(ConsoleAbortExc,         Exception);
./framework/uima/exceptions.hpp:  UIMA_EXC_CLASSDECLARE(ExcDocBuffer,            Uima_out_of_range);
./framework/uima/exceptions.hpp:  UIMA_EXC_CLASSDECLARE(ParseHandlerExc,         Exception);
./framework/uima/exceptions.hpp://? UIMA_EXC_CLASSDECLARE(ExcAccessError,      Uima_runtime_error);
./framework/uima/exceptions.hpp://? UIMA_EXC_CLASSDECLARE(ExcAssertionFailure, Exception);
./framework/uima/exceptions.hpp:  UIMA_EXC_CLASSDECLARE(ExcIllFormedInputError,  Uima_runtime_error);
./framework/uima/exceptions.hpp:  UIMA_EXC_CLASSDECLARE(ExcInvalidParameter,     Uima_invalid_argument);
./framework/uima/exceptions.hpp:  UIMA_EXC_CLASSDECLARE(ExcIndexOutOfRange,      Uima_out_of_range);
./framework/uima/exceptions.hpp://? UIMA_EXC_CLASSDECLARE(ExcDeviceError,      _runtime_error);
./framework/uima/exceptions.hpp:  UIMA_EXC_CLASSDECLARE(ExcInvalidRequest,       Uima_runtime_error);
./framework/uima/exceptions.hpp:  UIMA_EXC_CLASSDECLARE(ExcResourceExhausted,    Uima_runtime_error);
./framework/uima/exceptions.hpp:  UIMA_EXC_CLASSDECLARE(ExcOutOfMemory,          ExcResourceExhausted);
./framework/uima/exceptions.hpp://? UIMA_EXC_CLASSDECLARE(ExcOutOfSystemResource, ResourceExhausted);
./framework/uima/exceptions.hpp://? UIMA_EXC_CLASSDECLARE(ExcOutOfWindowResource, ResourceExhausted);
./framework/uima/exceptions.hpp:  UIMA_EXC_CLASSDECLARE(ExcFileNotFoundError,    Uima_runtime_error);
./framework/uima/exceptions.hpp:  UIMA_EXC_CLASSDECLARE(ExcWinCException,        Uima_runtime_error);
./framework/uima/exceptions.hpp:  UIMA_EXC_CLASSDECLARE(CodePageConversionException, Exception);
./framework/uima/exceptions.hpp:  UIMA_EXC_CLASSDECLARE(AprFailureException, Exception);
./framework/uima/exceptions.hpp:  To declare a new exception class, use the <TT>UIMA_EXC_CLASSDECLARE</TT> macro in an .h/.hpp
./framework/uima/exceptions.hpp:     UIMA_EXC_CLASSDECLARE(AnnotatorException, InvalidRequest);
./framework/uima/casiterator.hpp:  UIMA_EXC_CLASSDECLARE(CASIteratorException, uima::Exception);
./framework/uima/engine.hpp:  UIMA_EXC_CLASSDECLARE(UnknownTypeException, Exception);
./framework/uima/engine.hpp:  UIMA_EXC_CLASSDECLARE(UnknownFeatureException, Exception);
./framework/uima/engine.hpp:  UIMA_EXC_CLASSDECLARE(UnknownRangeTypeException, Exception);
./framework/uima/engine.hpp:  UIMA_EXC_CLASSDECLARE(IncompatibleRangeTypeException, Exception);
./framework/uima/engine.hpp:  UIMA_EXC_CLASSDECLARE(AllowedStringValuesIncompatibleException, Exception);
./framework/uima/engine.hpp:  UIMA_EXC_CLASSDECLARE(TypePriorityConflictException, Exception);
./framework/uima/engine.hpp:  UIMA_EXC_CLASSDECLARE(IncompatibleIndexDefinitionsException, Exception);
./framework/uima/engine.hpp:  UIMA_EXC_CLASSDECLARE(IncompatibleParentTypesException, Exception);
./framework/uima/engine.hpp:  UIMA_EXC_CLASSDECLARE(CASIncompatibilityException, Exception);
./framework/uima/engine.hpp:  UIMA_EXC_CLASSDECLARE(UimaAnalysisComponentException, Exception);
./framework/uima/internal_jedii_engine.hpp:    UIMA_EXC_CLASSDECLARE(JavaException, Exception);
./framework/uima/internal_capability_container.hpp:  UIMA_EXC_CLASSDECLARE(CapabilityException, Exception);
./framework/uima/taespecifierbuilder.hpp:  UIMA_EXC_CLASSDECLARE(InvalidXMLException, uima::Exception);
./framework/uima/caspool.hpp:  UIMA_EXC_CLASSDECLARE(CASPoolException, uima::Exception);
./framework/uima/taespecifier.hpp:  UIMA_EXC_CLASSDECLARE(ConfigParamLookupException, ConfigException);
./framework/uima/taemetadata.hpp:  UIMA_EXC_CLASSDECLARE(DuplicateConfigElemException, uima::Exception);
./framework/uima/taemetadata.hpp:  UIMA_EXC_CLASSDECLARE(ValidationException, uima::Exception);
./framework/uima/annotator.hpp:  UIMA_EXC_CLASSDECLARE(ExcEnumerationOverflow, ExcIllFormedInputError);
./framework/uima/config_param.hpp:  UIMA_EXC_CLASSDECLARE(ConfigException, Exception);
./framework/uima/config_param.hpp:  UIMA_EXC_CLASSDECLARE(ConfigParamException, ConfigException);
```

Reply:
Would you say UimaAnalysisComponentException is appropriate?